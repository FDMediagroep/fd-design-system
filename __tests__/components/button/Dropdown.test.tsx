import React from 'react';
import {
    Dropdown,
    getCssClassNames,
} from '../../../src/components/button/Dropdown';
import { render, fireEvent } from '@testing-library/react';

describe('Dropdown', () => {
    test('should render correctly', () => {
        const { getByText } = render(<Dropdown>Knopje</Dropdown>);

        expect(getByText('Knopje')).toBeTruthy();
    });

    test('should handle clicks correctly', () => {
        const spy = jest.fn();
        const { getByText, container } = render(
            <Dropdown onClick={spy}>Knopje</Dropdown>
        );

        expect(getByText('Knopje')).toBeTruthy();
        expect(spy).toBeCalledTimes(0);
        fireEvent.click(container.querySelector('button'));
        expect(spy).toBeCalledTimes(1);
        fireEvent.click(container.querySelector('button'));
        expect(spy).toBeCalledTimes(2);
    });

    test('should show dropdown correctly', () => {
        const { container, rerender, getByText } = render(
            <Dropdown>Knopje</Dropdown>
        );

        expect(
            container
                .querySelector('[aria-expanded]')
                .getAttribute('aria-expanded')
        ).toBe('false');

        fireEvent.click(container.querySelector('button'));

        expect(
            container
                .querySelector('[aria-expanded]')
                .getAttribute('aria-expanded')
        ).toBe('true');
    });

    test('should handle sizes correctly', () => {
        const { container } = render(<Dropdown size="s">Knopje</Dropdown>);
        const { container: container2 } = render(
            <Dropdown size="m">Knopje</Dropdown>
        );

        expect(container).toMatchSnapshot();
        expect(container2).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(<Dropdown>Knopje</Dropdown>);
        console.log(container.firstElementChild.classList);
        // expect(
        //     container?.firstElementChild?.classList.contains(
        //         getCssClassNames()[0]
        //     )
        // ).toBeTruthy();
    });
});
