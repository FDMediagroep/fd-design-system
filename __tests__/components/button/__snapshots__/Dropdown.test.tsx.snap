// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should handle sizes correctly 1`] = `
<div>
  <div
    class="fd-dropdown grid"
  >
    <button
      aria-expanded="false"
      class="xs-12  fd-dropdown-button"
    >
      <span
        class="xs__mr+1"
      >
        Knopje
      </span>
      <span
        class="collapse-icon"
      >
        <test-file-stub />
      </span>
      <span
        class="expand-icon"
      >
        <test-file-stub />
      </span>
    </button>
    <nav
      class="fd-dropdown-items"
    >
      <ul>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Spotify
              </span>
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Apple Podcasts
              </span>
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Google Podcasts
              </span>
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Dropdown should handle sizes correctly 2`] = `
<div>
  <div
    class="fd-dropdown grid"
  >
    <button
      aria-expanded="false"
      class="xs-12 m fd-dropdown-button"
    >
      <span
        class="xs__mr+1"
      >
        Knopje
      </span>
      <span
        class="collapse-icon"
      >
        <test-file-stub />
      </span>
      <span
        class="expand-icon"
      >
        <test-file-stub />
      </span>
    </button>
    <nav
      class="fd-dropdown-items"
    >
      <ul>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Spotify
              </span>
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Apple Podcasts
              </span>
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://fd.nl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                class="podcast-icon"
              >
                <test-file-stub />
              </span>
              <span>
                Google Podcasts
              </span>
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
